### PR TITLE
Av/28 initial gui for viewing data

### PR DESCRIPTION
**Pull request by anushkavidanage**
# Pull Request Details

## What issue does this PR address

- Initial GUI for viewing private data

- Link to associated issue: https://github.com/anusii/keypod/issues/28

## Checklist

[Use the check-list below to ensure your branch is ready for PR]

- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint errors (`make prep` or `flutter analyze lib`
                       python: use pre-commit ruff hook to check on commit)
- [x] No lint check errors related to your changes
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome
  - [ ] iOS
  - [x] Linux
  - [ ] MacOS
  - [x] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number
- [ ] Merge PR into dev

### CODE REVIEW

File: lib/src/screens/view_data/view_keys.dart
Comment (gjwgit) [inline lib/src/screens/view_data/view_keys.dart:1]: Why is this in here? Shouldn't this be in the keypod app?

**Review by gjwgit — CHANGES_REQUESTED** [review]
With KeyPod, click continue then display data (first time without a keypod folder) I get the red screen:

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following _TypeError was thrown building FutureBuilder<dynamic>(dirty, state:
_FutureBuilderState<dynamic>#5c59d):
type 'Null' is not a subtype of type 'String' in type cast

The relevant error-causing widget was:
  FutureBuilder<dynamic>
  FutureBuilder:file:///home/gjw/git/github/anusii/keypod/solidpod/lib/src/screens/view_data/view_keys.dart:69:13

When the exception was thrown, this was the stack:
#0      _ShowKeysState.build.<anonymous closure> (package:solidpod/src/screens/view_data/view_keys.dart:76:55)
#1      _FutureBuilderState.build (package:flutter/src/widgets/async.dart:609:55)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:5592:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5480:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5643:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:5196:7)
#6      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2904:19)
#7      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:989:21)
#8      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:448:5)
#9      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1386:15)
#10     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1311:9)
#11     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1169:5)
#12     _invoke (dart:ui/hooks.dart:312:13)
#13     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:399:5)
#14     _drawFrame (dart:ui/hooks.dart:283:31)

════════════════════════════════════════════════════════════════════════════════════════════════════
```



Testing with InnerPod, if I LOGIN then I see this as the app display.

image

That seems correct, and nice to see the addition of the back arrow somehow?

When I click the Continue button (named SESSION for InnerPod) I don't get the back arrow?

image


Was the back arrow added by the solidpod package? Should it be there in both scenarios.

This is my SolidLogin screen: 
image

File: lib/src/screens/view_data/view_keys.dart
Comment (gjwgit) [inline lib/src/screens/view_data/view_keys.dart:1]: The file name should match the main function name?

**Comment by anushkavidanage** [comment]
Fixed discussed issues. @gjwgit please have a look and let me know if you see any other issues.

**Review by gjwgit — APPROVED** [review]
Tested with KeyPod. All good.

**Comment by junhaow1** [comment]
will review after the standup

**Review by cdawei — APPROVED** [review]
Worked in my test using the corresponding branch of keypod.

**Review by junhaow1 — APPROVED** [review]
works well on my machine.

**Pull request merged**